core-library/src/main/java/org/silverpeas/core/io/media/image/imagemagick/Im4javaImageTool.java

```diff
@@ -149,4 +196,58 @@ private void resize(final IMOperation op,
       }
     }
   }
+
+  /**
+   * Centralizes text watermarking operation
+   * @param op
+   * @param source
+   * @param options
+   * @throws Exception
+   */
+  private void watermarkText(final IMOperation op, final File source,
+      final Map<Class<AbstractImageToolOption>, AbstractImageToolOption> options) throws Exception {
+    WatermarkTextOption watermarkText = getOption(options, WatermarkTextOption.class);
+    if (watermarkText != null) {
+      final String[] imageInfo;
+      DimensionOption dimension = getOption(options, DimensionOption.class);
+      if (dimension != null) {
+        imageInfo = new String[2];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)